ui/lib/client/index.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react'
+import i18next from 'i18next'
+import axios from 'axios'
+import { message, Modal, notification } from 'antd'
+import * as singleSpa from 'single-spa'
+
+import * as auth from '@lib/utils/auth'
+import * as routing from '@lib/utils/routing'
+import * as i18n from '@lib/utils/i18n'
+import publicPathPrefix from '@lib/utils/publicPathPrefix'
+
+import { DefaultApi } from './api'
+
+export * from './api'
+
+//////////////////////////////
+
+let basePath: string
+let apiClientInstance: DefaultApi
+
+function save(instanceBasePath: string, instance: DefaultApi) {
+  basePath = instanceBasePath
+  apiClientInstance = instance
+}
+
+function getInstance(): DefaultApi {
+  return apiClientInstance
+}
+
+function getBasePath(): string {
+  return basePath
+}
+
+export default { getInstance, getBasePath }
+
+//////////////////////////////
+
+export enum ErrorStrategy {
+  Default = 'default',
+  Custom = 'custom',
+}
+
+function initAxios() {
+  i18n.addTranslations(require.context('./translations/', false, /\.yaml$/))
+
+  const instance = axios.create()
+  instance.interceptors.response.use(undefined, function (err) {
+    const { response, config } = err
+    const errorStrategy = config.errorStrategy as ErrorStrategy
+    const method = (config.method as string).toLowerCase()
+
+    let errCode: string
+    if (err.message === 'Network Error') {
+      errCode = 'error.network'
+    } else {
+      errCode = response?.data?.code || err.message || 'error.api.other'
```

Comment:
In many cases, maybe displaying `response?.data?.message` is better since it contains verbose error details

Reply:
Maybe we can check whether the code is `error.api.other`. For this kind of error, display `data.message` instead.

Reply:
Maybe this check logic is also deserved to be shared with `ErrorBar`

Reply:
Make sense!